handlers/create_payable_resource.go

```diff
@@ -26,110 +26,131 @@ func CreatePayableResourceHandler(prDaoSvc dao.PayableResourceDaoService, apDaoS
 		log.DebugC(requestId, "successfully created payable resource", log.Data{"payable_resource": payableResource})
 
-		utils.WriteJSONWithStatus(w, r, payableResource, http.StatusCreated)
+		writeJSONResponse(w, r, payableResource, http.StatusCreated)
 
 		log.InfoC(requestId, "POST payable resource request completed successfully")
 	})
 }
 
-func extractRequestData(w http.ResponseWriter, r *http.Request, err error, request models.PayableRequest) (authentication.AuthUserDetails, string, string, bool) {
+// decodeRequest decodes the request body into PayableRequest struct
+func decodeRequest(r *http.Request) (models.PayableRequest, error) {
+	var request models.PayableRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	return request, err
+}
+
+// extractRequestData extracts auth details, company code and penalty ref type from the request/context
+func extractRequestData(w http.ResponseWriter, r *http.Request, request models.PayableRequest) (authentication.AuthUserDetails, string, string, bool) {
 	var authUserDetails authentication.AuthUserDetails
-	// request body failed to get decoded
-	if err != nil {
-		log.ErrorR(r, fmt.Errorf("invalid request"))
-		m := models.NewMessageResponse("failed to read request body")
-		utils.WriteJSONWithStatus(w, r, m, http.StatusBadRequest)
-		return authUserDetails, "", "", true
-	}
 
 	userDetailsValue := r.Context().Value(authentication.ContextKeyUserDetails)
 	if userDetailsValue == nil {
 		log.ErrorR(r, fmt.Errorf("user details not in context"))
-		m := models.NewMessageResponse("user details not in request context")
-		utils.WriteJSONWithStatus(w, r, m, http.StatusBadRequest)
+		writeJSONResponse(w, r, models.NewMessageResponse("user details not in request context"), http.StatusBadRequest)
 		return authUserDetails, "", "", true
 	}
 
 	companyCode, err := getCompanyCodeFromTransaction(request.Transactions)
 	if err != nil {
 		log.ErrorR(r, fmt.Errorf("company code cannot be resolved"))
-		m := models.NewMessageResponse("company code cannot be resolved")
-		utils.WriteJSONWithStatus(w, r, m, http.StatusBadRequest)
+		writeJSONResponse(w, r, models.NewMessageResponse("company code cannot be resolved"), http.StatusBadRequest)
 		return authUserDetails, "", "", true
 	}
 
 	penaltyRefType, err := getPenaltyRefTypeFromTransaction(request.Transactions)
 	if err != nil {
 		log.ErrorR(r, fmt.Errorf("penalty reference type cannot be resolved"))
-		m := models.NewMessageResponse("penalty reference type cannot be resolved")
-		utils.WriteJSONWithStatus(w, r, m, http.StatusBadRequest)
+		writeJSONResponse(w, r, models.NewMessageResponse("penalty reference type cannot be resolved"), http.StatusBadRequest)
 		return authUserDetails, "", "", true
 	}
 
 	authUserDetails = userDetailsValue.(authentication.AuthUserDetails)
 	return authUserDetails, companyCode, penaltyRefType, false
 }
+
+// validateTransactions checks that transactions are valid payable penalties
```

Comment:
```suggestion
// validateTransactions to ensure that the transactions in the request are valid payable penalties that exist in E5
```
My suggestion is based on the previous comment that mentions exist in E5 might be worth keeping here?